funcs_scripts.php

```diff
@@ -350,21 +406,21 @@ function human_cron(string $cron): string
  * between 1 and 28
  * Note that this will return the exact same value every time it is called for a given filename in a given module
  */
-function predictable_random_int($max, $offset = 0): int
+function predictable_random_int($scriptName, $max, $offset = 0): int
 {
-    global $MODULE_DIR;
-    $callingFile = debug_backtrace()[0]['file'];
-    // remove absolute path e.g. /home/myuser/...
-    $moduleStandardiserDir = basename(__DIR__);
-    $dirQuoted = preg_quote($moduleStandardiserDir);
-    // double $dirQuoted is for github actions CI where there which will have a directory strcture
-    // with /module-standardiser/module-standardiser/...
-    if (!preg_match("#/$dirQuoted/$dirQuoted/(.+)$#", $callingFile, $matches)) {
-        preg_match("#/$dirQuoted/(.+)$#", $callingFile, $matches);
-    }
-    $relativePath = $matches[1];
-    $chars = str_split("$MODULE_DIR-$relativePath");
+    $chars = str_split(module_name() . $scriptName);
     $codes = array_map(fn($c) => ord($c), $chars);
-    mt_srand(array_sum($codes));
-    return mt_rand(0, $max) + $offset;
+    $sum = array_sum($codes);
+    $remainder = $sum % ($max + 1);
+    return $remainder + $offset;
+}
+
+/**
+ * Determine if the current brain is either main or master
+ */
+function current_branch_is_non_semver()
+{
+    global $MODULE_DIR;
+    $currentBranch = cmd('git rev-parse --abbrev-ref HEAD', $MODULE_DIR);
+    return in_array($currentBranch, ['main', 'master']);
```

Comment:
Do a regex instead. We know what semver looks like, and we also know that there are other non-semver branches we might end up with. E.g. if the standardiser crashes midway through making PRs we might end up in a `pulls/` style branch.

Reply:
Updated and changed to `current_branch_is_semver()`